addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/JSFunctionPassiveScanner.java

```diff
@@ -0,0 +1,205 @@
+import net.htmlparser.jericho.Source;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+/** Passive Scan Rule for Dangerous JS Functions https://github.com/zaproxy/zaproxy/issues/5673 */
+public class JSFunctionPassiveScanner extends PluginPassiveScanner {
+
+    /** Prefix for internationalized messages used by this rule */
+    private static final String MESSAGE_PREFIX = "pscanalpha.jsfunction.";
+
+    public static final String FUNC_LIST_DIR = "xml";
+    public static final String FUNC_LIST_FILE = "js-function-list.txt";
+    private static final Logger LOGGER = Logger.getLogger(JSFunctionPassiveScanner.class);
+    private static final int PLUGIN_ID = 10110;
+
+    public static final List<String> DEFAULT_FUNCTIONS = Collections.emptyList();;
+    private static final Supplier<Iterable<String>> DEFAULT_PAYLOAD_PROVIDER =
+            () -> DEFAULT_FUNCTIONS;
+    public static final String JS_FUNCTION_PAYLOAD_CATEGORY = "JS-Function";
+
+    private static Supplier<Iterable<String>> payloadProvider = DEFAULT_PAYLOAD_PROVIDER;
+
+    private static List<Pattern> patterns = null;
+    private PassiveScanThread parent = null;
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+        // This rule only scans responses received
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        if (!msg.getResponseHeader().isHtml() && !msg.getResponseHeader().isJavaScript()) {
+            return;
+        }
+        if (patterns == null) {
+            patterns = getPatterns();
+        }
+        StringBuilder evidence = new StringBuilder();
+        if (msg.getResponseBody().length() > 0 && msg.getResponseHeader().isHtml()) {
+            // Check the scripts in HTML
+            Element el;
+            int offset = 0;
+            while ((el = source.getNextElement(offset, HTMLElementName.SCRIPT)) != null) {
+                String elStr = el.toString();
+                for (Pattern pattern : patterns) {
+                    if (pattern.matcher(elStr).find()) {
+                        evidence.append(elStr);
+                        evidence.append("\n");
+                        break; // Only need to record this script once
+                    }
+                }
+                offset = el.getEnd();
+            }
+        } else if (msg.getResponseBody().length() > 0 && msg.getResponseHeader().isJavaScript()) {
+            String[] lines = msg.getResponseBody().toString().split("\n");
+            for (String line : lines) {
+                for (Pattern pattern : patterns) {
+                    if (pattern.matcher(line).find()) {
+                        evidence.append(
```

Comment:
Same comment regarding the evidence (this might be better for the other info field too).

Reply:
You mean split the call to `Constant.messages.getString(MESSAGE_PREFIX + "otherinfo", pattern, line));` into a  separate variable to be displayed as other info rather than have it in the evidence?

Reply:
Yeah, but we could still use part of the "line" as evidence.

Reply:
I decided on removing the  "otherinfo" altogether as I didn't think it was necessary